File: testdata/fixtures/sample.R
library(dplyr)
library(ggplot2)

analyze_data <- function(df) {
  result <- df %>%
    filter(value > 0) %>%
    group_by(category) %>%
    summarize(mean = mean(value))
  return(result)
}

plot_results <- function(data) {
  ggplot(data, aes(x = category, y = mean)) +
    geom_bar(stat = "identity")
}

DataProcessor <- function(config) {
  self <- list(
    config = config,
    data = NULL
  )
  class(self) <- "DataProcessor"

  self$load <- function(file) {
    self$data <- read.csv(file)
    return(self)
  }

  self$process <- function() {
    analyze_data(self$data)
  }

  return(self)
}

processor <- DataProcessor(list(verbose = TRUE))
processed <- processor$process()

File: testdata/fixtures/sample.hs
module DataProcessor where

import Data.List (sort, group)
import qualified Data.Map as Map

-- | Calculate statistics from a list of numbers
calculateStats :: [Int] -> (Int, Int, Int)
calculateStats xs = (minimum xs, maximum xs, sum xs)

-- | Process data with configuration
processData :: Config -> [Int] -> [Int]
processData config xs =
    let filtered = filter (> 0) xs
    in sort filtered

-- | Main data processor function
runProcessor :: String -> IO ()
runProcessor inputFile = do
    contents <- readFile inputFile
    let stats = calculateStats (map read (lines contents))
    print stats

-- | Data configuration type
data Config = Config {
    verbose :: Bool,
    threshold :: Int
} deriving (Show, Eq)

-- | Create a new config
createConfig :: Bool -> Int -> Config
createConfig v t = Config v t
